File: src/main/java/com/insogroup/utils/initialisator/interfaces/Initializable.java
package com.insogroup.utils.initialisator.interfaces;

import com.insogroup.utils.initialisator.exceptions.InitializableException;
import org.json.JSONObject;

/**
 * Objects which implement (or extend) this interface must be initialized by calling method @Initializable.init
 */
public interface Initializable {
    /**
     * Init the object
     * @param initJSON setting for initialisation
     * @throws InitializableException Throws when object can't be initialised with current input json
     */
    void init(JSONObject initJSON) throws InitializableException;
}

File: src/main/kotlin/com/insogroup/utils/initialisator/interfaces/Initializable.kt
package com.insogroup.utils.initialisator.interfaces

import com.github.insanusmokrassar.iobjectk.interfaces.IObject
import com.insogroup.utils.initialisator.exceptions.InitializableException

interface Initializable {
    @Throws(InitializableException::class)
    fun initObject(from: IObject<Any>)
}